src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,370 @@
+            s_connectionListField = s_connectionGroupType?.GetField("m_ConnectionList", BindingFlags.Instance | BindingFlags.NonPublic);
+            s_connectionType = systemNetHttpAssembly?.GetType("System.Net.Connection");
+            s_writeListField = s_connectionType?.GetField("m_WriteList", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            // Second step: Generate an accessor for HttpWebRequest._HttpResponse
+            FieldInfo field = typeof(HttpWebRequest).GetField("_HttpResponse", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            string methodName = field?.ReflectedType.FullName + ".get_" + field.Name;
+            if (!string.IsNullOrEmpty(methodName))
+            {
+                DynamicMethod getterMethod = new DynamicMethod(methodName, typeof(HttpWebResponse), new Type[] { typeof(HttpWebRequest) }, true);
+                ILGenerator generator = getterMethod.GetILGenerator();
+                generator.Emit(OpCodes.Ldarg_0);
+                generator.Emit(OpCodes.Ldfld, field);
+                generator.Emit(OpCodes.Ret);
+                s_httpResponseAccessor = (Func<HttpWebRequest, HttpWebResponse>)getterMethod.CreateDelegate(typeof(Func<HttpWebRequest, HttpWebResponse>));
+            }
+
+            // Double checking to make sure we have all the pieces initialized
+            if (s_connectionGroupListField == null ||
+                s_connectionGroupType == null ||
+                s_connectionListField == null ||
+                s_connectionType == null ||
+                s_writeListField == null ||
+                s_httpResponseAccessor == null)
+            {
+                throw new InvalidOperationException("Unable to initialize all required reflection objects");
+            }
+        }
+
+        private static void PerformInjection()
+        {
+            FieldInfo servicePointTableField = typeof(ServicePointManager).GetField("s_ServicePointTable", BindingFlags.Static | BindingFlags.NonPublic);
+            if (servicePointTableField == null)
+            {
+                throw new InvalidOperationException("Unable to access the ServicePointTable field");
+            }
+
+            Hashtable originalTable = servicePointTableField.GetValue(null) as Hashtable;
+            ServicePointHashtable newTable = new ServicePointHashtable();
+
+            // Copy the existing entries over to the new table, and replace the field with our new table
+            if (originalTable != null)
+            {
+                foreach (DictionaryEntry entry in originalTable)
+                {
+                    newTable[entry.Key] = entry.Value;
+                }
+            }
+
+            servicePointTableField.SetValue(null, newTable);
+        }
+
+#endregion
+
+        internal static HttpHandlerDiagnosticListener s_instance = new HttpHandlerDiagnosticListener();
+
+#region private fields
+        private const string DiagnosticListenerName = "System.Net.Http.Desktop.V4";
```

Comment:
I think we should drop the .V4 suffix.    We expect this to work for all Desktop frameworks for all time.

(I looked to see what the 'official' moniker was for Desktop and it is 'net'. which sadly more confusing than enlightening, so I guess we eek the '.Desktop'